pkg/oci/load_balancer_spec.go

```diff
@@ -161,21 +173,36 @@ func (s *LBSpec) Certificates() (map[string]loadbalancer.CertificateDetails, err
 	if s.SSLConfig == nil {
 		return certs, nil
 	}
-
-	cert, key, err := s.SSLConfig.readSSLSecret(s.service)
+	//Read listener Kubernetes Secret
+	sslSecret, err := s.SSLConfig.readSSLSecret(s.service.Namespace, s.SSLConfig.ListenerSSLSecretName)
 	if err != nil {
-		return nil, errors.Wrap(err, "reading SSL Secret")
+		return nil, errors.Wrap(err, "reading SSL Listener Secret")
 	}
-	if cert == "" || key == "" {
+	if len(sslSecret.PublicCert) == 0 || len(sslSecret.PrivateKey) == 0 {
```

Comment:
I'm not clear on how this works in the case that there's BackendSet certs but not Listener?

Reply:
LMAATCFR (let me add a test case for that) good spot